### PR TITLE
feat: add caption data to modal images

### PR DESCRIPTION
**Pull request by colincclark**
https://nidigitalsolutions.jira.com/browse/REPLAT-2477
https://nidigitalsolutions.jira.com/browse/REPLAT-2456 (android)
https://nidigitalsolutions.jira.com/browse/REPLAT-2989 (iOS)

- added caption/credits to the modal image
- ensure close button is positioned as per designs (right on Android and left on iOS)
- passed caption/credit data through article lead and inline images
- fixed up modal storybook to better showcase the modal (with knobs for caption data)

Future PR to come will fix other issues mentioned at the bottom of the JIRA ticket:
- https://nidigitalsolutions.jira.com/browse/REPLAT-3315
- https://nidigitalsolutions.jira.com/browse/REPLAT-3316

### CODE REVIEW

File: packages/image/src/modal-image.js [hunk 1]
```diff
@@ -27,9 +30,11 @@ class ModalImage extends Component {
   }
 
   render() {
+    const { caption, credits } = this.props;
+
     const closeButton = (
       <Svg height={24} viewBox="0 0 24 24" width={24}>
-        <G fill={colours.functional.cancel}>
+        <G fill="#FFFFFF">
```
Comment (craigbilner) [inline packages/image/src/modal-image.js]: this should still come from the styleguide though?

File: yarn.lock [hunk 2]
```diff
@@ -937,6 +937,13 @@
     fs-extra "6.0.0"
     glob "7.1.2"
 
+"@times-components/caption@2.1.1":
```
Comment (craigbilner) [inline yarn.lock]: got dep issues

File: packages/article/src/article-lead-asset/article-lead-asset-image.js [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
-export const propTypes = {
+ArticleLeadAssetImage.propTypes = {
+  caption: PropTypes.string,
+  credits: PropTypes.string,
   crop: cropPropTypes.isRequired
 };
 
-ArticleLeadAssetImage.propTypes = propTypes;
+ArticleLeadAssetImage.defaultProps = {
+  caption: "",
```
Comment (craigbilner) [inline packages/article/src/article-lead-asset/article-lead-asset-image.js]: I think if we use default to represent the absence of something `null` is favourable over empty string?

Comment (colincclark) [reply packages/article/src/article-lead-asset/article-lead-asset-image.js]: that feels odd to me, but have amended to `null`

File: yarn.lock (hunk 2 above)
Comment (colincclark) [reply yarn.lock]: yep, fixed dep

File: packages/image/src/modal-image.js (hunk 1 above)
Comment (colincclark) [reply packages/image/src/modal-image.js]: done

File: packages/styleguide/src/colours/functional.js [hunk 4]
```diff
@@ -14,7 +14,8 @@ const functionalColours = {
   articleFlagExclusive: "#C51D24",
   articleFlagNew: "#E34605",
   articleFlagUpdated: "#3C81BE",
-  cancel: "rgba(255, 255, 255, 0.8)"
+  cancel: "rgba(255, 255, 255, 0.8)",
```
Comment (tuncaulubilge) [inline packages/styleguide/src/colours/functional.js]: Looks like we are not using this cancel colour anywhere else, we can ditch it

Comment (colincclark) [reply packages/styleguide/src/colours/functional.js]: removed

**Pull request merged**